native/explorer/src/lazyframe.rs

```diff
@@ -72,3 +72,35 @@ pub fn lf_slice(data: ExLazyFrame, offset: i64, length: u32) -> Result<ExLazyFra
     let lf = data.clone_inner();
     Ok(ExLazyFrame::new(lf.slice(offset, length)))
 }
+
+#[rustler::nif]
+pub fn lf_filter_with(data: ExLazyFrame, ex_expr: ExExpr) -> Result<ExLazyFrame, ExplorerError> {
+    let ldf = data.clone_inner();
+    let expr = ex_expr.clone_inner();
+
+    Ok(ExLazyFrame::new(ldf.filter(expr)))
+}
+
+#[rustler::nif]
+pub fn lf_filter_with_aggregation(
+    data: ExLazyFrame,
+    ex_expr: ExExpr,
+    groups: Vec<&str>,
+) -> Result<ExLazyFrame, ExplorerError> {
+    let ldf = data.clone_inner();
+    let aggs: Vec<Expr> = ldf
+        .schema()?
+        .iter_names()
+        .filter(|name| !groups.contains(&name.as_str()))
+        .map(|name| col(name).filter(ex_expr.clone_inner()).list().keep_name())
```

Comment:
I think we may be able to break it apart.

For example, when we call group in the lazy frame, we emit `ldf.groupby_stable`. Then when we call filter, we transform the aggregations, and so on. And we explode only at the end when we ungroup. Ping me today and we can talk. :)

Reply:
Just to keep track: we decided to not support the groupby with aggregations for this PR. I reverted in https://github.com/elixir-nx/explorer/pull/476/commits/a6f4b761a3b82c0368aea2282794f77ff77455b7